edb/edgeql/compiler/config.py

```diff
@@ -292,11 +292,20 @@ def _validate_config_object(
 def _validate_global_op(
     expr: qlast.ConfigOp, *, ctx: context.ContextLevel
 ) -> SettingInfo:
+    glob_name = s_utils.ast_ref_to_name(expr.name)
     glob = ctx.env.get_schema_object_and_track(
-        s_utils.ast_ref_to_name(expr.name), expr.name,
+        glob_name, expr.name,
         modaliases=ctx.modaliases, type=s_globals.Global)
     assert isinstance(glob, s_globals.Global)
 
+    if isinstance(expr, (qlast.ConfigSet, qlast.ConfigReset)):
+        if glob.get_expr(ctx.env.schema):
+            raise errors.ConfigurationError(
+                f"global '{glob_name}' is computed from an expression and "
```

Comment:
I'd probably just say "computed global 'whatever' may not be modified" or some such